tests/system/test_signals.c

```diff
@@ -19,7 +19,19 @@
 /* cause a general fault protection */
 static void child_generate_gpf(void *unused)
 {
+#if defined(__i386__) || defined(__x86_64__)
+
    asmVolatile("hlt");
+
+#elif defined(__riscv)
+/*
+ * The GPF type exception does not exist in riscv, and if executing a
+ * privileged instruction in user space then it generates an illegal
+ * instruction exception.
+ * Here we access the zero address to generate a page exception.
+ */
+   *(int *)0 = 1;
```

Comment:
That might work now.. but it's completely UB to de-reference the NULL pointer in C. So.. you'd need to do a bit of "compiler magic" here to get the same effect in a UB-free way. I'm sure you'll figure out a way do it. One safe approach is using memcpy() to do that. Another is a cast with `volatile` but that might not be enough if the compiler "sees" the actual address. 

Note that I also have UBSAN builds and tests. So.. those will fail when we enable them for RISCV. Maybe play a bit with UBSAN on a regular userspace program and check what appears to be safe.

Reply:
Use the memory load statement `lw t0, (zero)` instead (zero is a general-purpose register that is always equals 0).